master/internal/resourcemanagers/provisioner/provisioner.go

```diff
@@ -113,20 +113,21 @@ func (p *Provisioner) provision(ctx *actor.Context) {
 		ctx.Log().WithError(err).Error("cannot list instances")
 		return
 	}
-	if p.scaleDecider.updateInstanceSnapshot(instances) {
+	updated := p.scaleDecider.updateInstanceSnapshot(instances)
+	if updated {
 		ctx.Log().Infof("found state changes in %d instances: %s",
 			len(instances), fmtInstances(instances))
-
-		if p.scaleDecider.db != nil {
-			err = p.scaleDecider.recordInstanceStats(p.SlotsPerInstance())
-			if err != nil {
-				ctx.Log().WithError(err).Error("cannot record instance stats")
-			}
-		}
 	}
 
 	p.scaleDecider.calculateInstanceStates()
 
+	if updated && p.scaleDecider.db != nil {
```

Comment:
i'd rather the nil checks be closer to the db access. e.g., in the event someone were to call `updateInstanceStartStats` without remembering to check.

Reply:
i assume it's this way for tests or something, but still

Reply:
also this is a comment without much context on the pr so prob just ignore me

Reply:
I think this makes sense, fixed it, thank you.